src/lib.rs

```diff
@@ -440,7 +440,13 @@ impl RotatingFile {
             let Some(to_remove) = file_history.pop_front() else {
                 break;
             };
-            if let Err(error) = std::fs::remove_file(&to_remove) {
+            if let Err(error) = std::fs::metadata(&to_remove).and_then(|meta| {
+                if meta.is_dir() {
+                    std::fs::remove_dir_all(&to_remove)
```

Comment:
Hmmm.

Convince me that if a directory is here (where *we don't expect it* and *we don't know how it got here, but we know it wasn't ours*), the correct course of action is to unilaterally remove it. Is there any chance this is an honest mistake and we just blew away something that actually matters?

Reply:
I had to think about this for a while before I really hit on the convincing argument here.

These files are not simply generated by a `fs::read_dir()` on the root dir. Well, they are, but it's a _filtered_ one that we do on startup: a random directory found inside the root dir would only be added to the file history (candidates for future removal) if we were told to use `CleanupStrategy::All` when constructing the `RotatingFile`. That option, as opposed to `CleanupStrategy::MatchingFormat`, is an explicit acknowledgement that foreign items in the root directory are _always_ okay to remove, even if their names don't match the expected format for log files.

In our case, because our usage of this crate is tightly constrained and we control the root directory, we can confidently say that if any directories are found in the root dir at startup, they are unwanted and we're okay to remove them. I don't believe, at the moment, we're planning on upstreaming changes to this crate, but even if someone other than us did use this code, they would have to have passed `CleanupStrategy::All` to explicitly ask for extraneous files to be removed.

Reply:
I am convinced!